lib/WP_Auth0_DBManager.php

```diff
@@ -290,6 +290,56 @@ public function install_db( $version_to_install = null, $app_token = '' ) {
 			}
 		}
 
+		// 3.6.0
+
+		if ( ( $this->current_db_version < 18 && 0 !== $this->current_db_version ) || 18 === $version_to_install ) {
+
+			// Migrate passwordless_method
+			if ( $options->get( 'passwordless_enabled', FALSE ) ) {
+				$pwl_method = $options->get( 'passwordless_method' );
+				switch ( $pwl_method ) {
+
+					// SMS passwordless just needs 'sms' as a connection
+					case 'sms' :
+						$options->set( 'lock_connections', 'sms' );
+						break;
+
+					// Social + SMS means there are existing social connections we want to keep
+					case 'socialOrSms' :
+						$options->add_lock_connection( 'sms' );
+						break;
+
+					// Email link passwordless just needs 'email' as a connection
+					case 'emailcode' :
+					case 'magiclink' :
+						$options->set( 'lock_connections', 'email' );
+						break;
+
+					// Social + Email means there are social connections be want to keep
+					case 'socialOrMagiclink' :
+					case 'socialOrEmailcode' :
+						$options->add_lock_connection( 'email' );
+						break;
+				}
+
+				// Need to set a special passwordlessMethod flag if using email code
+				if ( in_array( $pwl_method, array( 'emailcode', 'socialOrEmailcode' ) ) ) {
```

Comment:
If I understand correctly, this `if` is checking if the selected type wishes a code instead of a link. Because you're relying on the auth0.js default value (currently link as you pointed out) this thing only changes that value if the type is "code". What if tomorrow that library changes the default?? ---> You should add an `else` to this `if` that sets the "link" type.


```php
$lock_json = trim( $options->get( 'extra_conf' ) );
$lock_json_decoded = ! empty( $lock_json ) ? json_decode( $lock_json, TRUE ) : array();
$lock_json_decoded[ 'passwordlessMethod' ] = in_array( $pwl_method, array( 'emailcode', 'socialOrEmailcode' ) ) ?  'code' : 'link'; //or however it's called
$options->set( 'extra_conf', json_encode( $lock_json_decoded ) );
```